source/FreeRTOS_DNS_Parser.c

```diff
@@ -373,124 +420,189 @@
                             {
-                                if( xBufferAllocFixedSize == pdFALSE )
-                                {
-                                    size_t uxDataLength = uxBufferLength +
-                                                          sizeof( UDPHeader_t ) +
-                                                          sizeof( EthernetHeader_t ) +
-                                                          sizeof( IPHeader_t );
+                                break;
+                            }
+                        }
 
-                                    /* Set the size of the outgoing packet. */
-                                    pxNetworkBuffer->xDataLength = uxDataLength;
-                                    pxNewBuffer = pxDuplicateNetworkBufferWithDescriptor( pxNetworkBuffer,
-                                                                                          uxDataLength +
-                                                                                          sizeof( LLMNRAnswer_t ) );
+                        pxEndPoint = pxNetworkBuffer->pxEndPoint;
 
-                                    if( pxNewBuffer != NULL )
-                                    {
-                                        BaseType_t xOffset1, xOffset2;
+                        /* Make a copy of the end-point because xApplicationDNSQueryHook() is allowed
+                         * to write into it. */
+                        ( void ) memcpy( &( xEndPoint ), pxEndPoint, sizeof( xEndPoint ) );
 
-                                        xOffset1 = ( BaseType_t ) ( pucByte - pucUDPPayloadBuffer );
-                                        xOffset2 = ( BaseType_t ) ( ( ( uint8_t * ) pcRequestedName ) - pucUDPPayloadBuffer );
+                        #if ( ipconfigUSE_IPv6 != 0 )
+                            {
+                                /*logging*/
+                                FreeRTOS_printf( ( "prvParseDNS_HandleLLMNRRequest[%s]: type %04X\n", pxSet->pcName, pxSet->usType ) );
 
-                                        pxNetworkBuffer = pxNewBuffer;
-                                        pucNewBuffer = &( pxNetworkBuffer->pucEthernetBuffer[ ipUDP_PAYLOAD_OFFSET_IPv4 ] );
+                                xEndPoint.usDNSType = xSet.usType;
+                            }
+                        #endif /* ( ipconfigUSE_IPv6 != 0 ) */
 
-                                        pucByte = &( pucNewBuffer[ xOffset1 ] );
-                                        pcRequestedName = ( char * ) &( pucNewBuffer[ xOffset2 ] );
-                                        pxDNSMessageHeader = ( ( DNSMessage_t * ) pucNewBuffer );
+                        /* If this is not a reply to our DNS request, it might an LLMNR
+                         * request. */
+                        if( xApplicationDNSQueryHook( &xEndPoint, xSet.pcName ) )
+                        {
+                            != pdFALSE )
```

Comment:
Something went wrong here.

The old code went like:
~~~
    if( xApplicationDNSQueryHook( &xEndPoint, pxSet->pcName ) == pdFALSE )
    {
        /* This device doesn't have this name. */
        break;
    }
~~~
Here an LLMNR request was received. The application looks up `pcName` and returns false if the name is not for this device.

Reply:
As there is no while case here, so we are not using break. Hence the condition is negated and checked. If **!= pdFalse** then we are doing all the things.